udp-echo/src/main.rs

```diff
@@ -0,0 +1,47 @@
+extern crate actix;
+extern crate bytes;
+extern crate futures;
+extern crate tokio;
+
+use actix::{Actor, AsyncContext, Context, Message, StreamHandler};
+use bytes::BytesMut;
+use futures::stream::SplitSink;
+use futures::{Future, Sink, Stream};
+use std::net::SocketAddr;
+use tokio::codec::BytesCodec;
+use tokio::net::{UdpFramed, UdpSocket};
+
+struct UdpActor {
+    sink: SplitSink<UdpFramed<BytesCodec>>,
+}
+impl Actor for UdpActor {
+    type Context = Context<Self>;
+}
+
+#[derive(Message)]
+struct UdpPacket(BytesMut, SocketAddr);
+impl StreamHandler<UdpPacket, std::io::Error> for UdpActor {
+    fn handle(&mut self, msg: UdpPacket, _: &mut Context<Self>) {
+        println!("Received: ({:?}, {:?})", msg.0, msg.1);
+        (&mut self.sink).send((msg.0.into(), msg.1)).wait().unwrap();
```

Comment:
If I'm not mistaken `Stream::send` should consume self? Does it compile?

Reply:
It is `Sink::send` not a `Stream::send`.
https://docs.rs/futures/0.1.25/futures/sink/trait.Sink.html#method.send

Reply:
Yeah ok, it doesn't change the fact that it takes `self` rather than `&mut self`

Reply:
Interesting. But this code works somehow. :(

Reply:
Ok, the travis checks passed showing that this code compiles. Should I fix it if it works? I do not like the situation when I do not understand why it works and will be happy to find explanation, but I definitely need a help.

Reply:
If it compiles then everything is ok I guess. which is surprising but fine too